### PR TITLE
[OptionsResolver] add `ignoreUndefined()`

### LINKED ISSUE #18282: [OptionsResolver] add `ignoreUndefined()` method to allow skip not inte…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#49608
| PR author(s) | @Constantine Shtompel
| Merged in    | 6.3

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 6.3 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by alamirault**
Fix #18282

### CODE REVIEW

File: components/options_resolver.rst [hunk 1]
```diff
@@ -842,6 +842,29 @@ the option::
 This closure receives as argument the value of the option after validating it
 and before normalizing it when the option is being resolved.
 
+Ignore not defined Options
```
Comment (alamirault) [inline components/options_resolver.rst:845]: Sometime there an uppercase, sometime not.
Can you explain when we have to use it ?

Thanks

Comment (javiereguiluz) [reply components/options_resolver.rst:845]: It's called "title case" or "headline case" and the rules are "complicated" (i.e. we can't define a rule). There are several styles. Here you can see some examples: https://en.wikipedia.org/wiki/Title_case

File: components/options_resolver.rst [hunk 2]
```diff
@@ -842,6 +842,29 @@ the option::
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, all options are resolved, validated and an :class:`Symfony\\Component\\OptionsResolver\\Exception\\UndefinedOptionsException`
+is thrown if an unknown option is passed. You can ignore not defined options by using the
+:method:`Symfony\\Component\\OptionsResolver\\OptionsResolver::ignoreUndefined` method::
+
+    // ...
+    $resolver
+        ->setDefined(['hostname'])
+        ->ignoreUndefined(true)
```
Comment (yceruto) [inline components/options_resolver.rst]: It should be `setIgnoreUndefined(true)` here. Only the fluent configurator has this `ignoreUndefined()` method

**Review by yceruto — CHANGES_REQUESTED** [review]

File: components/options_resolver.rst (hunk 2 above)
Comment (alamirault) [reply components/options_resolver.rst]: Oops, fixed !

**Review by OskarStark — APPROVED** [review]

**Review by yceruto — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Antoine.

**Pull request merged**